x-pack/auditbeat/module/system/socket/state.go

```diff
@@ -869,20 +862,56 @@ func (f *flow) updateWith(ref flow, s *state) {
 	f.remote.updateWith(ref.remote)
 }
 
-func (s *state) onFlowTerminated(f *flow) {
+func (s *state) reportFlow(f *flow) {
+	if f != nil && f.isValid() && int(f.pid) != s.currentPID {
+		ev, err := f.toEvent(true)
+		if err != nil {
+			s.log.Errorf("Failed to convert flow=%v err=%v", f, err)
+			return
+		}
+		s.reporter.Event(ev)
+	}
+}
+
+func (s *state) reportFlows(l *linkedList) (count int) {
+	for head := l.head; head != nil; head = head.Next() {
+		if f, ok := head.(*flow); ok {
+			s.reportFlow(f)
+			count++
```

Comment:
If `*state.reportFlow` returns a conditional 1 for success and 0 for failure then the count would more accurately reflect what was reported (depending on whether you are interested in the attempts or the succeeds).

Reply:
makes sense, done.